### PR TITLE
Added an option to query metrics based on tag only (regardless to type)

### LINKED ISSUE #175: Add an option to query metrics based on tag only (regardless to type)
```ruby
Hawkular::Metrics::Client.new(entrypoint, credentials, options).strings.get_data_by_tags(:miq_metric => true)
```

will work, but i need it to work without the strings section.
cc @pilhuhn @zgalor @simon3z.

### PR DESCRIPTION
**Pull request by rubenvp8510**
Fixes #175

Also, added buckets parameter to get_data_by_tags method.

### CODE REVIEW

File: lib/hawkular/metrics/metric_api.rb [hunk 1]
```diff
@@ -191,19 +203,15 @@ def raw_data(ids, starts: nil, ends: nil, limit: nil, order: nil)
...
+      def get_data_by_tags(tags, starts: nil, ends: nil, bucketDuration: nil, buckets:nil)
+        params = { tags: @client.tags_param(tags), start: starts,
+                   end: ends, bucketDuration: bucketDuration, buckets: buckets }
         path = "/#{@resource}/"
         @legacy_api ? path << 'data/?' : path << 'stats/?'
         resp = @client.http_get(path + encode_params(params))
         resp.is_a?(Array) ? resp : [] # API returns no content (empty Hash) instead of empty array
       end
 
-      def tags_param(tags)
```
Comment (josejulio) [inline lib/hawkular/metrics/metric_api.rb:203]: What is the reason to move it above ?

Comment (rubenvp8510) [reply lib/hawkular/metrics/metric_api.rb:203]: I need to use it at the level of the Client class. so I moved it to that class in order to reuse the code.

File: lib/hawkular/metrics/metric_api.rb [hunk 2]
```diff
@@ -12,6 +12,18 @@ def default_timestamp(array)
       array
     end
 
+    def tags_param(tags)
+      tags.map { |k, v| "#{k}:#{v}" }.join(',')
+    end
+
+    def get_all_data_by_tags(tags, buckets: nil, bucketDuration:nil, start:nil, ends: nil)
```
Comment (josejulio) [inline lib/hawkular/metrics/metric_api.rb]: Could we use data_by_tags instead? It seems to be more ruby-like.

File: lib/hawkular/metrics/metric_api.rb [hunk 3]
```diff
@@ -191,19 +203,15 @@ def raw_data(ids, starts: nil, ends: nil, limit: nil, order: nil)
       # @param bucketDuration [String] optional interval (default no aggregation)
       # @return [Array[Hash]] datapoints
       # @see #push_data #push_data for datapoint detail
-      def get_data_by_tags(tags, starts: nil, ends: nil, bucketDuration: nil)
-        params = { tags: tags_param(tags), start: starts,
-                   end: ends, bucketDuration: bucketDuration }
+      def get_data_by_tags(tags, starts: nil, ends: nil, bucketDuration: nil, buckets:nil)
```
Comment (josejulio) [inline lib/hawkular/metrics/metric_api.rb:214]: Could you document the buckets param?

File: lib/hawkular/metrics/metric_api.rb (hunk 2 above)
Comment (rubenvp8510) [reply lib/hawkular/metrics/metric_api.rb]: You're right, I'll rename this method.

Comment (josejulio) [reply lib/hawkular/metrics/metric_api.rb]: While you are at it, could you also document the method and its params ?

File: lib/hawkular/metrics/metric_api.rb (hunk 3 above)
Comment (josejulio) [reply lib/hawkular/metrics/metric_api.rb:214]: Also, maybe (in a separate-PR) we should deprecate this `get_data_by_tags` and call it just `data_by_tags`... (using an alias).

**Comment by josejulio** [comment]
LGTM

**Comment by simon3z** [comment]
@yaacov this may be interesting for you as well.

**Pull request merged**